jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -283,8 +284,7 @@ public boolean offer(Runnable task)
 
             try
             {
-                _task.put(task);
-                return true;
+                return _task.offer(task, 1, TimeUnit.SECONDS);
```

Comment:
If this times out, then we don't put the reserved thread back onto the stack, so the reserved thread may then wait forever.  We need to at least flag that it has to die.

Reply:
Maybe we should never use a blocking-without-timeout read call to the `SynchronousQueue`?